decryptor/mysql/prepared_statements_test.go

```diff
@@ -35,3 +42,171 @@ func TestNewMysqlCopyTextBoundValue(t *testing.T) {
 		}
 	})
 }
+
+// columnPacketHex is mysql test column packet with Name `id` and table `test_type_aware_decryption_without_defaults`
+var columnPacketHex = "0364656604746573742b746573745f747970655f61776172655f64656372797074696f6e5f776974686f75745f64656661756c74732b746573745f747970655f61776172655f64656372797074696f6e5f776974686f75745f64656661756c74730269640269640c3f000b000000030342000000"
+
+// paramPacketHex is mysql test param packet with Name `?` and table `test_type_aware_decryption_without_defaults`
+var paramPacketHex = "03646566000000013f000c3f0000000000fd8000000000"
+
+func TestColumnsTrackHandler(t *testing.T) {
+	data, err := hex.DecodeString(columnPacketHex)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	parser := sqlparser.New(sqlparser.ModeStrict)
+
+	testConfig := `
+schemas:
+  - table: test_type_aware_decryption_without_defaults
+    columns:
+      - id
+    encrypted:
+      - column: id
+        data_type: "str"
+`
+	schemaStore, err := config.MapTableSchemaStoreFromConfig([]byte(testConfig))
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	setting := base.NewProxySetting(parser, schemaStore, nil, nil, nil, nil, false)
+	proxyHandler, err := NewMysqlProxy(&stubSession{}, parser, setting)
+	if err != nil {
+		t.Fatal()
+	}
+
+	fieldTracker := NewPreparedStatementFieldTracker(proxyHandler, 1)
+
+	fieldPacket := NewPacket()
+	fieldPacket.SetData(data)
+
+	server, client := net.Pipe()
+	defer client.Close()
+
+	wg := sync.WaitGroup{}
+	wg.Add(1)
+	// starting server connection read goroutine
+	go func() {
+		defer wg.Done()
+		resData := make([]byte, len(fieldPacket.data)+len(fieldPacket.header))
+		_, err := server.Read(resData)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		resPacket := NewPacket()
+		// without header which is 4 bytes
+		resPacket.SetData(resData[4:])
+		resDesc, err := ParseResultField(resPacket)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		if resDesc.Type != TypeString {
+			t.Fatalf("result packet type should be %d (string) but was %d", TypeString, resDesc.Type)
+		}
+	}()
+
+	err = fieldTracker.ColumnsTrackHandler(context.Background(), fieldPacket, server, client)
+	if err != nil {
+		t.Fatal(err)
+	}
+	wg.Wait()
```

Comment:
what if something went wrong and `wg.Done()` wasn't call?)

Reply:
Added time out to Pipe connection, more over we have `defer wg.Done` so in case of any fatal it will be called.